tests/src/test/scala/org/http4s/MessageSpec.scala

```diff
@@ -62,6 +62,15 @@ class MessageSpec extends Http4sSpec {
         Request(Method.GET).addCookie("token", "value").headers.get("Cookie".ci).map(_.value) must beSome("token=value")
       }
     }
+
+    "Request.copy" should {
+      "update pathInfo after Request uri is changed" in {
```

Comment:
Not changing it to def makes sense.

Thanks, I'll modify it to work with the caret

Is this what you mean by null initialization in ```Header.parses``` ?

https://github.com/http4s/http4s/blob/588690d0574aa33671660a19fc78b60dab1feab5/core/src/main/scala/org/http4s/Header.scala#L86